extensions/mssql/src/credentialstore/credentialstore.ts

```diff
@@ -37,7 +40,87 @@ class CredentialsFeature extends SqlOpsFeature<any> {
 		});
 	}
 
-	protected registerProvider(options: any): Disposable {
+	/**
+	 * Removes a credential for a given connection profile from sqltoolsservice
+	 * and adds it to keytar
+	 */
+	private async cleanCredential(conn: azdata.connection.ConnectionProfile): Promise<boolean> {
+		const credentialId = Utils.formatCredentialId(conn);
+		// read password from every saved password connection
+		const credential = await this._client.sendRequest(Contracts.ReadCredentialRequest.type, { credentialId, password: undefined });
+		if (credential.password) {
+			const password = credential.password;
+			// save it in keychain
+			await this.keychain.setPassword(credentialId, password);
+			// check if it's saved
+			const savedPassword = await this.keychain.getPassword(credentialId);
+			if (savedPassword === password) {
+				// delete from tools service
+				const result = await this._client.sendRequest(Contracts.DeleteCredentialRequest.type,
+					{ credentialId, password: undefined });
+				return result;
+			} else {
+				return false;
+			}
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Try to migrate all the credentials to the keytar system
+	 * @returns
+	 */
+	private migrateCredentials(): Thenable<boolean> {
```

Comment:
Why are you using Thenable? Should use Promises for anything that isn't on the extensibility layer.

And while you're at it make this async so it's much easier to read.

Reply:
This is an internal function though - there's no reason to use thenable for those.